docs/docs/reference/experimental/canthrow.md

```diff
@@ -71,19 +71,32 @@ How can the ability be produced? There are several possibilities:
 Most often, the ability is produced by having a using clause `(using CanThrow[Exc])` in some enclosing scope. This roughly corresponds to a `throws` clause
 in Java. The analogy is even stronger since alongside `CanThrow` there is also the following type alias defined in the `scala` package:
 ```scala
-infix type canThrow[R, +E <: Exception] = CanThrow[E] ?=> R
+infix type $throws[R, +E <: Exception] = CanThrow[E] ?=> R
 ```
-That is, `R canThrow E` is a context function type that takes an implicit `CanThrow[E]` parameter and that returns a value of type `R`. Therefore, a method written like this:
+That is, `R $throws E` is a context function type that takes an implicit `CanThrow[E]` parameter and that returns a value of type `R`. What's more, the compiler
+will translate an infix types with `throws` as the operator to `$throws` applications
+according to the rules
+```
+                A throws E  -->  A $throws E
+    A throws E₁ | ... | Eᵢ  -->  A $throws E₁ ... $throws Eᵢ
```

Comment:
Personally, I find it a little bit confusing that `A throws E1 | E2` does not translate to `CanThrow[E1 | E2] ?=> A`. What would be the problem of using such a desugaring?

Reply:
We don't have a rule for fulfilling a CanThrow[E1 | E2] contract. Logically, it can be fulfilled only if we have both a CanThrow[E1] and a CanThrow[E2], which is what the rewriting achieves.

Reply:
Thinking more about it…
From the user of the capability, we have the choice of using one of the given capabilities, so indeed the syntax `A throws E1 | E2` makes sense because in the implementation we can choose to throw `E1` _or_ `E2`.
On the other hand, from the point of view of the caller, we need to provide both the capabilities to throw `E1` _and_ `E2` (so, eventually, we a context parameter of type `CanThrow[E1 & E2]`, or two context parameters `CanThrow[E1]` and `CanThrow[E2]`).
So, I’m really not sure about overloading the `|` operator with an _ad-hoc_ meaning just for the case of exceptions. Maybe we should just use a comma?

~~~ scala
def f(): A throws E1, E2
// syntax for:
def f()(using CanThrow[E1], CanThrow[E2]): A
~~~

Also, how would such a syntax generalize to other types of capabilities?

About the rule for summoning `CanThrow[E1 & E2]`, would it make sense to change the current rule that summons one `CanThrow[E]` per exception handler in a `try` expression to summon one `CanThrow[E1 & E2 & ...]` per `try` expression? Or, introduce something like the following?

~~~
given [E1: CanThrow, E2: CanThrow]: CanThrow[E1 & E2] = ???
~~~

Reply:
Using a comma would work if the `throws` clause was only permitted as a method result type. But in fact, it's used as an infix type that can also appear in other contexts. And there  comma would be ambiguous.
```scala
 given [E1: CanThrow, E2: CanThrow]: CanThrow[E1 & E2] = ???
```
That might be a possibility. We need to be careful not to introduce ambiguities with this rule, though.